pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix

```diff
@@ -88,12 +88,34 @@ self: super: {
   ##       from the context: a
   constraints = super.constraints_0_10;
 
+  ## Needs bump to a versioned attribute
+  ## 
+  ##     • Could not deduce (Semigroup (IterT m a))
+  ##         arising from the superclasses of an instance declaration
+  ##       from the context: (Monad m, Monoid a)
+  free = super.free_5;
+
+  funcmp = overrideCabal super.funcmp_1_9 (drv: {
+    ## Needs bump to a versioned attribute
+    ## 
+    ## Needed for (<>) in prelude
+    ## Setup: Encountered missing dependencies:
+    ## base >=3 && <4.11
+    jailbreak       = true;
```

Comment:
I am surprised that jailbreak is necessary. I released funcmp 1.9 specifically to address ghc 8.4.x compatibility. Did I make a mistake somewhere? According to http://hackage.haskell.org/package/funcmp, the 1.9 version accepts ` base >= 4.5 && < 5`.

Reply:
Something must have messed up in the gearbox, investigating..

Reply:
What I ran to facilitate investigation:
```
nh audit --skip-baseline
Generating override cache: 63/63
Assembling overrides................................................................. done.
                  ChasingBottoms: no upstream specified
                      haskell-gi: hackage 0.21.0 obsoletes upstream fix (version 0.21.0), committed Tue Jan  9 19:38:03 MSK 2018 < release uploaded Thu, 11 Jan 2018 09:57:02 GMT
                 haskell-gi-base: hackage 0.21.0 obsoletes upstream fix (version 0.21.0), committed Tue Jan  9 19:38:03 MSK 2018 < release uploaded Thu, 11 Jan 2018 09:55:04 GMT
               vector-algorithms: no upstream specified
deepfire@andromedae:~/holotype$ nh trim --skip-baseline funcmp
Generating override cache: 63/63
-( trimming in progress
TRIM funcmp>
TRIM funcmp -all>  Assembling overrides................................................................. done.
FATAL: failed to build 'funcmp', without overriding 'default'
FATAL: logs:  nh log funcmp default
CABAL-MISSING-DEPS
Setup: Encountered missing dependencies:
base >=3 && <4.11
FAILED
TRIM funcmp -jailbreak>    ## Needs bump to a versioned attribute
  ##
  ## Needed for (<>) in prelude  funcmp = super.funcmp_1_9;

Assembling overrides................................................................. done.
FATAL: failed to build 'funcmp', without overriding 'jailbreak'
FATAL: logs:  nh log funcmp jailbreak
CABAL-MISSING-DEPS
Setup: Encountered missing dependencies:
base >=3 && <4.11
FAILED
TRIM funcmp -src>  SKIPPED, explanation exists:  nh x emit_explanation src funcmp
deepfire@andromedae:~/holotype$ nh info funcmp
attribute 'funcmp':

           status: shadowed
        overrides: jailbreak src
  nixpkgs release: 1.8
 upstream release: 1.9
 override release: 1.9
             meta: src.explanation
```
What I see from the above, is that https://raw.githubusercontent.com/NixOS/nixpkgs/master/pkgs/development/haskell-modules/hackage-packages.nix still defines `funcmp` to be 1.8 -- which still has the problem for any default users of `funcmp` on GHC 8.4 -- even if they wouldn't have it if they used `funcmp_1_9`.

Reply:
cc @peti

Reply:
The `without overriding 'default'` "explanation" is sufficiently confusing it needs attention, I guess..  It actually means "without overriding _anything_" or "by _default_"..

Reply:
Well, `funcmp` does not compile on GHC 8.4.x even after the jailbreak `funcmp_1_9` compiles fine on GHC 8.4.x without the jailbreak. Adding the jailbreak to the 1.9 version does not hurt, but it seems unnecessary to me (and it's not going to benefit users of the 1.8 version in any way, either).

Reply:
You're right, the trimmer has messed up and was building `funcmp-1.8` while attempting to trim the jailbreak override -- that's clearly a bug..